src/components/spreadsheet/data-fetching/use-fetch-equipment.ts

```diff
@@ -0,0 +1,104 @@
+                (node) =>
+                    node.id === nodeId && (node.type === NodeType.ROOT || isStatusBuilt(node.data?.globalBuildStatus))
+            ) !== undefined,
+        [treeNodes]
+    );
+
+    const fetchNodesEquipementData = useCallback(
+        (nodeIds: Set<string>, onFetchingDone: undefined | ((info: FetchingInfo) => void)) => {
+            if (studyUuid && currentRootNetworkUuid && currentNodeUuid) {
+                let fetcherPromises: Promise<unknown>[] = [];
+                let spreadsheetEquipmentsByNodes: SpreadsheetEquipmentsByNodes = {
+                    nodesId: [],
+                    equipmentsByNodeId: {},
+                };
+                let errorMessage: string | null = null;
+
+                nodeIds.forEach((nodeId) => {
+                    if (currentNodeUuid === nodeId || isBuilt(nodeId)) {
+                        const promise = getFetcher(type)(studyUuid, nodeId as UUID, currentRootNetworkUuid, []);
+                        fetcherPromises.push(promise);
+                        promise
+                            .then((results) => {
+                                let fetchedEquipments = results.flat();
+                                spreadsheetEquipmentsByNodes.nodesId.push(nodeId);
+                                fetchedEquipments = formatEquipments(fetchedEquipments);
+                                spreadsheetEquipmentsByNodes.equipmentsByNodeId[nodeId] = fetchedEquipments;
+                            })
+                            .catch((err) => {
+                                console.error(
+                                    `Fetching error for type ${type.toString()} on node ${nodeId} (${err.message})`
+                                );
+                            });
+                    } else {
+                        // mark unbuilt node as processed with empty entry/result
+                        spreadsheetEquipmentsByNodes.nodesId.push(nodeId);
+                        spreadsheetEquipmentsByNodes.equipmentsByNodeId[nodeId] = [];
+                    }
+                });
+
+                Promise.all(fetcherPromises)
+                    .then(() => {
+                        dispatch(loadEquipments(type, spreadsheetEquipmentsByNodes));
+                        console.debug(
+                            `Equipment data fetching and dispatch done for ${fetcherPromises.length} built nodes among ${nodeIds.size}`
+                        );
+                    })
+                    .catch((err) => {
+                        console.debug('Equipment data fetching and dispatch NOT done');
+                        errorMessage = err.message;
+                    })
+                    .finally(() => {
+                        if (onFetchingDone) {
+                            onFetchingDone({
+                                errorMessage: errorMessage,
+                            });
+                        }
+                    });
```

Comment:
```suggestion
                    .catch((err: unknown) => {
                        console.debug('Equipment data fetching and dispatch NOT done');
                        if (onError) {
                            catchErrorHandler(err, onError);
                        }
                    });
```
why not something like that instead ?